include/deal.II/lac/arpack_solver.h

```diff
@@ -139,13 +157,20 @@ class ArpackSolver : public Subscriptor
 
   /**
    * Solve the generalized eigensprectrum problem $A x=\lambda B x$ by calling
-   * the <code>dneupd</code> and <code>dnaupd</code> functions of ARPACK.
+   * the <code>dsaupd</code> and <code>dseupd</code> or
+   * <code>dnaupd</code> and <code>dneupd</code> functions of ARPACK.
    *
    * The function returns a vector of eigenvalues of length <i>n</i> and a
-   * vector of eigenvectors, where the latter should be twice the size of the
-   * eigenvalue vector. The first <i>n</i> vectors in
-   * <code>eigenvectors</code> will be the real parts of the eigenvectors, the
-   * second <i>n</i> the imaginary parts.
+   * vector of eigenvectors of length <i>n</i> in the symmetric case
+   * and of length <i>n+1<i> in the non-symmetric case. In the symmetric case
```

Comment:
Suppose the eigenvalues 1 and 2 are real and 3 and 4 are a complex conjugate pair.
Asking for 3 eigenpairs results in the storage scheme
[real(vector1),real(vector2),real(vector3),imag(vector3)]
So that we need `n+1` columns to store the complex values of the 3rd vector. 
Note that we have the same storage scheme in this case if we ask for 4 eigenpairs.
The point is that we don't know in advance whether we choose n such that complex conjugate pairs are kept together or as in the given example are separated by `n`.
Only in the case If the `n`-th eigenvalue is real then we only need length `n`.